### PR TITLE
New Method + property + change in backup() method

### PR DESCRIPTION
**Pull request by LxLeChat**
- Added a new method called SetLogRotation to set hidden property LogRotation to the [HostsFile] Class
- Added a new hidden property called ComputerName to the [HostsFile] Class.
- Modified [HostsFile] Class constructors to attribute a value to the ComputerName Property
- All this changes to a prepare the new Save-HFMHostsFile.

- Change the backup method that accepts a system.io.directoryinfo as a parameter

### CODE REVIEW

File: HostsFileManagement/Classes/Public/HostsFile.ps1 [hunk 1]
```diff
@@ -1,24 +1,26 @@
...
-      $this.Path ="\\$env:Computername\admin$\System32\drivers\etc\hosts"
+      $This.Path ="\\$env:Computername\admin$\System32\drivers\etc\hosts"
+      $This.ComputerName = $env:Computername
     }
     
     HostsFile([String]$ComputerName){
     
       if (Test-Connection -ComputerName $ComputerName -Quiet -Count 2){
         $This.Path ="\\$Computername\admin$\System32\drivers\etc\hosts"
+        $This.ComputerName = $env:Computername
```
Comment (Stephanevg) [inline HostsFileManagement/Classes/Public/HostsFile.ps1]: This should be:

```powershell
$This.ComputerName = $Computername
```

**Review by Stephanevg — CHANGES_REQUESTED** [review]
Hi, there is a copy / paste issue on line 17. Other then that, it looks good!

**Comment by LxLeChat** [comment]
you are right !

**Review by Stephanevg — APPROVED** [review]
All good!

**Comment by Stephanevg** [comment]
Awesome! thanks once more @LxLeChat !

**Pull request merged**